### ISSUE TITLE
Remove definition of _availability_version_check

### ISSUE THREAD

**Issue by zanderso** [flutter/flutter#133203]
To fix https://github.com/flutter/flutter/issues/132130, a strong definition of `_availability_version_check` was added to avoid the weak linkage of that symbol coming from the clang runtime library. The weak linkage of that symbol would cause apps to be rejected by the app store. When Clang 18 becomes the version used by Xcode, presumably the App Store will not have a problem with the symbol anymore, and we can therefore remove the file `flutter/shell/platform/darwin/common/availability_version_check.cc`, which was added in https://github.com/flutter/engine/pull/44711.

**Comment by chinmaygarde** [comment]
I don't think we can do this until some time next year till Xcode is updated.
